### PR TITLE
Update for new CI mechanism

### PR DESCRIPTION
**Pull request by SimonHeybrock**
- PR and main builds do not use `conda build`
- Packages use `conda build` for entire process.
- Packages will be made nightly (publish to `dev`) and from tags (publish to `main`).

### CODE REVIEW

File: stages.yml [hunk 1]
```diff
@@ -47,20 +47,12 @@ parameters:
 - name: noarch
   type: boolean
   default: false
-- name: build_package
```
Comment (nvaytet) [inline stages.yml:50]: so for the `ess-notebooks`, we were not building a C++ library and neither were we building a conda package nor running python tests.
Just building docs (which is why I had this parameter).
Not sure how we best handle the `ess-notebooks` builds. We basically just want to do a `sphinx-build`.

Comment (SimonHeybrock) [reply stages.yml:50]: So basically we *do* deploy, but only docs? And we want to use `sphinx-build` rather than `conda build`?

Comment (SimonHeybrock) [reply stages.yml:50]: I suppose the easiest option may be to figure out the details once we roll this out to `ess-notebooks`?

Comment (nvaytet) [reply stages.yml:50]: Yes, there is nothing to 'conda build'. It just runs the `build_docs.py` script.
The tests is just to check that notebooks actually run without errors.

File: tools/build_cpp.py [hunk 2]
```diff
@@ -63,9 +63,8 @@ def main(prefix='install', build_dir='build', source_dir='.', caching=False):
                 '-DCMAKE_OSX_DEPLOYMENT_TARGET':
                 osxversion,
                 '-DCMAKE_OSX_SYSROOT':
-                os.path.join('/Applications', 'Xcode.app', 'Contents',
-                             'Developer', 'Platforms', 'MacOSX.platform',
-                             'Developer', 'SDKs',
+                os.path.join('/Applications', 'Xcode.app', 'Contents', 'Developer',
```
Comment (nvaytet) [inline tools/build_cpp.py:66]: This long path was removed entirely from the `scippneutron` PR. Can we remove it here also?

Comment (SimonHeybrock) [reply tools/build_cpp.py]: Done

**Pull request merged**